Impossible-Agent-Suite/DangerousActivities/src/darklist.verifier.spec.ts

```diff
@@ -0,0 +1,79 @@
+import { createAddress } from "forta-agent-tools";
+import DarklistVerifier, { dataUrl } from "./darklist.verifier";
+
+describe("DarklistVerifier tests suite", () => {
+  const mockFetcher = jest.fn();
+  let verifier: DarklistVerifier;
+
+  const verifyCalls = (numberOfCalls: number) => {
+    expect(mockFetcher).toHaveBeenCalledTimes(numberOfCalls);
+    for(let i = 1; i <= numberOfCalls; ++i)
+      expect(mockFetcher).nthCalledWith(i, dataUrl);
+  };
+
+  const toResponse = (addrs: string[]) => {
+    return {
+      data: addrs.map(addr => {
+        return {
+          "address": addr,
+        };
+      }),
+    };
+  };
+
+  beforeEach(() => {
+    mockFetcher.mockClear();
+    verifier = new DarklistVerifier(20, mockFetcher);
+  });
+
+  it("should set the threshold correctly", () => {
+    for(let i = 1; i <= 20; ++i){
+      const verifier = new DarklistVerifier(i, mockFetcher);
+      expect(verifier.threshold).toStrictEqual(i);
+    }
+  });
+
+  it("should return True only for the addresses in the blacklist", async () => {
+    const address: string[] = [
+      createAddress('0xdead'),
+      createAddress('0xfee'),
+      createAddress('0xdef1'),
+      createAddress('0xe0a'),
+      createAddress('0x5eed'),
+    ];
```

Comment:
would be better to change the variable name to addresses instead of address, same in the following test cases.